cpp/src/arrow/io/memory_benchmark.cc

```diff
@@ -152,6 +154,99 @@ static void StreamReadWrite(void* src, void* dst, size_t size) {
   }
 }
 
+#endif  // ARROW_HAVE_SSE4_2
+
+#ifdef ARROW_HAVE_ARMV8_CRYPTO
+
+using VectorType = uint8x16_t;
+using VectorTypeDual = uint8x16x2_t;
+
+#define VectorSet vdupq_n_u8
+#define VectorLoadAsm vld1q_u8
+
+static void armv8_stream_load_pair(VectorType* src, VectorType* dst) {
+  asm volatile("LDNP %[reg1], %[reg2], [%[from]]\n\t"
+               : [reg1] "+r"(*dst), [reg2] "+r"(*(dst + 1))
+               : [from] "r"(src));
+}
+
+static void armv8_stream_store_pair(VectorType* src, VectorType* dst) {
+  asm volatile("STNP %[reg1], %[reg2], [%[to]]\n\t"
+               : [to] "+r"(dst)
+               : [reg1] "r"(*src), [reg2] "r"(*(src + 1))
+               : "memory");
+}
+
+static void armv8_stream_ldst_pair(VectorType* src, VectorType* dst) {
+  asm volatile(
+      "LDNP q1, q2, [%[from]]\n\t"
+      "STNP q1, q2, [%[to]]\n\t"
+      : [from] "+r"(src), [to] "+r"(dst)
+      :
+      : "memory", "v0", "v1", "v2", "v3");
+}
+
+static void Read(void* src, void* dst, size_t size) {
+  const auto simd = static_cast<uint8_t*>(src);
```

Comment:
Thanks for your comments.
Simplify implementation of `Read` in new commit.

For definition of `VectorLoadAsm` :
```
vld1q_u8 (const uint8_t *__a)
{
  return (uint8x16_t)
    __builtin_aarch64_ld1v16qi ((const __builtin_aarch64_simd_qi *) __a);
}

```
The input parameter type is `uint8_t *`. 
So I cast it to `uint8_t *`  rather than `VectorType*` to avoid to duplicate casting from `VectorType*` to `uint8_t*`.